cpp/test/distance/masked_nn.cu

```diff
@@ -314,8 +314,8 @@ template <typename K, typename V, typename L>
                                        cudaStream_t stream = 0)
 {
   typedef typename raft::KeyValuePair<K, V> KVP;
-  std::shared_ptr<KVP> exp_h(new KVP[size]);
-  std::shared_ptr<KVP> act_h(new KVP[size]);
+  std::shared_ptr<KVP[]> exp_h(new KVP[size]);
+  std::shared_ptr<KVP[]> act_h(new KVP[size]);
```

Comment:
This is needed so that `new[]` matches with `delete[]`. Without this, the deleter uses `delete` instead of `delete[]` and compilation fails.